test/com/google/enterprise/adaptor/DocIdSenderTest.java

```diff
@@ -491,6 +534,56 @@ public void testAclItemToString() {
     assertEquals(golden, "" + new DocIdSender.AclItem(id, acl));
   }
 
+
+  private static Map<GroupPrincipal, Collection<Principal>> groupsSample() {
+    // Order of iteration matters
+    Map<GroupPrincipal, Collection<Principal>> groups
+        = new TreeMap<GroupPrincipal, Collection<Principal>>();
+    groups.put(new GroupPrincipal("g1"),
+        Arrays.asList(new UserPrincipal("u1"), new GroupPrincipal("g2")));
+    groups.put(new GroupPrincipal("g2"),
+        Arrays.asList(new UserPrincipal("u2"), new GroupPrincipal("g3")));
+    groups.put(new GroupPrincipal("g3"),
+        Arrays.asList(new UserPrincipal("u3"), new GroupPrincipal("g4")));
+    return Collections.unmodifiableMap(groups);
+  }
+
+  private static List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>>
+      unsplitExpectedResult() {
+    List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>> goldenGroups
+        = new ArrayList<List<Map.Entry<GroupPrincipal,
+          Collection<Principal>>>>();
+    List<Map.Entry<GroupPrincipal, Collection<Principal>>> tmp
+        = new ArrayList<Map.Entry<GroupPrincipal, Collection<Principal>>>();
+    tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
```

Comment:
I get this was ugly before. My two thoughts to make this more readable, which may not be necessary, because the contents are pretty straightfoward, are to add a newEntry helper method to infer the generic types, or to do something like:

new ArrayList<ugly type>(ImmutableMap.of(...).entrySet())

ImmutableMap uses insertion order for iteration.